test/lib/usd/translators/testUsdImportInstances.py

```diff
@@ -0,0 +1,570 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import maya.api.OpenMaya as om
+import os
+import unittest
+
+from maya import cmds
+from maya import standalone
+
+from pxr import Kind
+from pxr import Usd
+from pxr import UsdGeom
+
+import fixturesUtils
+
+class testUsdImportInstances(unittest.TestCase):
+    """Test for importing instances. The test cases used here were designed
+    by pmolodo as future improvements on the import/export of instances. We
+    expect them to load without errors, but not necessarily to provide perfect
+    results at this point in time."""
+    @classmethod
+    def setUpClass(cls):
+        cls._path = fixturesUtils.setUpClass(__file__)
+
+    @classmethod
+    def tearDownClass(cls):
+        standalone.uninitialize()
+
+    def sortedPathsTo(self, node):
+        """Get a sorted list of paths going to the specified node"""
+        sel = om.MGlobal.getSelectionListByName(node)
+        self.assertEqual(sel.length(), 1)
+        val = [str(i) for i in om.MDagPath.getAllPathsTo(sel.getDependNode(0))]
+        val.sort()
+        return tuple(val)
+
+    def testImportSimpleUnoptimized(self):
+        """
+        Test importing the following non-optimized USD simple instancing
+        scenario:
+
+        pCube1 [Xform]
+            pCubeShape1 [Scope] (instance /InstanceSources/pCube1_pCubeShape1)
+        pCube2 [Xform]
+            pCubeShape1 [Scope] (instance /InstanceSources/pCube1_pCubeShape1)
+        pCube3 [Xform]
+            pCubeShape1 [Scope] (instance /InstanceSources/pCube1_pCubeShape1)
+        InstanceSources
+            pCube1_pCubeShape1 [Scope]
+                pCubeShape1 [Mesh]
+
+        The goal would be for Maya to skip all [Scope] items and end up with:
+
+        pCube1 [transform]
+            pCubeShape1 [mesh] (instance "source")
+        pCube2 [transform]
+            pCubeShape1 [mesh] (instance pCube1|pCubeShape1)
+        pCube3 [transform]
+            pCubeShape1 [mesh] (instance pCube1|pCubeShape1)
+
+        But that will happen in a second phase. We only make sure this imports
+        without coding errors here.
+        """
+        usd_file = os.path.join(self._path, "UsdImportInstancesTest",
+                                "SimpleScenario_unoptimized.usda")
+
+        cmds.file(f=True, new=True)
+        cmds.mayaUSDImport(file=usd_file,
+                           primPath="/",
+                           instanceMode="buildInstances")
```

Comment:
I think this setting of `primPath` can be removed, since that should be the default? Similarly, `instanceMode=buildInstances` looks like it was made the default as well, but I'm ok with that staying since I think it makes the intent of the test clear.

Reply:
Isn't primPath the `defaultPrim`, which would be pCube1 in the test scenes?